packages/mdc-card/_mixins.scss

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2017 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+@import "@material/theme/mixins";
+
+//
+// Public
+//
+
+@mixin mdc-card-fill-color($color) {
+  @include mdc-theme-prop(background-color, $color);
+}
+
+@mixin mdc-card-corner-radius($radius) {
+  border-radius: $radius;
+}
+
+@mixin mdc-card-media-aspect-ratio($x, $y) {
+  &::before {
+    // This clever trick brought to you by: http://www.mademyday.de/css-height-equals-width-with-pure-css.html
+    margin-top: percentage($y / $x);
+  }
+}
+
+//
+// Private
+//
+
+@mixin mdc-card-box-sizing_ {
+  &,
+  &::before,
```

Comment:
Do we really need the `::before` and `::after` selectors on every single thing this mixin is applied to? I would only expect it to be needed on before/after for `__action` and possibly `card` itself assuming we have ripples on the whole card at some point.

Reply:
There's no direct need for it ATM, but I wanna try to start consistently applying `border-box` to all of our elements (and pseudo elements) so that there are no surprises in the future (for us _or_ our clients).

I can't tell you how many times I've gone to add `padding` to something in MDC Web and had it end up bigger than I expected because of stupid `content-box`. I want to get to a place where we can assume that all of our elements are `border-box` by default, unless we specifically override it for some reason.

WDYT?

Reply:
I guess, but most of our elements don't have `::before` and `::after` unless they have ripples, which is why I thought that in particular seemed excessive. Keep in mind how much CSS this is adding because we're including this mixin multiple times.

Maybe add a parameter to this mixin to conditionalize the inclusion of before/after in the selector? (There's an example of how to conditionalize the selector in the focus state styles in mdc-ripple/_mixins)

Reply:
As discussed offline: Libraries/frameworks generally should not mess with `::before` and `::after` unless absolutely necessary (e.g., mdc-ripple). This is to prevent conflicts with other libraries and client code which might _also_ set styles on the pseudos.

Removed the `mdc-card-box-sizing_` mixin and directly applied `box-sizing: border-box` to each element individually.